src/record/key.rs

```diff
@@ -196,13 +196,19 @@ impl RemoteId {
     }
 
     /// Create a new `local` [`RecordId`].
-    pub fn local(sub_id: &str) -> Self {
+    pub fn local(sub_id: &str) -> Result<Self, RecordError> {
         let mut full_id = String::with_capacity(9);
```

Comment:
Haha the 9 is a bit random (my fault), maybe a nice round number like `16` is more sensible? I guess we need at least 6 bytes for the `local:`, and then I feel like 10 bytes is a good amount of space for most cases to avoid realloc.

Reply:
Nice round number like 0b10000! Done.

Reply:
Feels like unnecessary optimization anyway, mainly I got slightly bothered by the 9.

Reply:
Great thanks!